src/main/java/dmytro/hadiuchko/springboot/entity/User.java

```diff
@@ -25,4 +37,48 @@ public class User {
     private String lastName;
     @Column(name = "shipping_address")
     private String shippingAddress;
+    @ManyToMany(fetch = FetchType.EAGER)
+    @JoinTable(
+            name = "users_roles",
+            joinColumns = @JoinColumn(name = "user_id"),
+            inverseJoinColumns = @JoinColumn(name = "role_id")
+    )
+    private Set<Role> roles;
+    @Column(name = "is_deleted")
+    private boolean isDeleted = false;
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return roles;
+    }
+
+    @Override
+    public String getUsername() {
+        return email;
+    }
+
+    @Override
+    public String getPassword() {
+        return password;
+    }

```

Comment:
I am not sure but Data annotation will create getter for Passord field, I think if we haven't got password field we should override getPassword method and return credentials we are using for auth